code/game/machinery/vending/vending.dm

```diff
@@ -107,7 +107,7 @@
 		var/price = prices[typepath]
 		if(isnull(amount)) amount = 1
 
-		var/obj/item/temp_path = typepath
+		var/obj/item/temp_path = new typepath
```

Comment:
Stop instancing things that dont need to be instanced

Reply:
It's either instancing or snowflake code that handles seeds specifically :/